### ISSUE TITLE
Type serialization

### ISSUE THREAD

**Issue by zojize** [runtypes/runtypes#236]
As typescript currently cannot support [complex type serialization](http://blog.wolksoftware.com/decorators-metadata-reflection-in-typescript-from-novice-to-expert-part-4#4-complex-types-serialization_1), it would be nice if runtypes could emit a serialized type definitions, and I could use it for auto-generated API documentation and other uses cases involved in telling the user what inputs I want to take in. For example:
```ts
import { Record, Number, String } from 'runtypes';

const Foo = Record({
    bar: Number,
    baz: String,
});

console.log(Foo.serialized());
```

and the code above would emit something like:

```javascript

// borrowed syntax from http://blog.wolksoftware.com/decorators-metadata-reflection-in-typescript-from-novice-to-expert-part-4#4-complex-types-serialization_1
{
     kind: "interface",
     members: {
         bar: Number,
         baz: String,
     }
}
```

And this information could be helpful for the use case stated above and maybe other use cases of choice. Right now, I am not sure how to you could represent unions, template literal types or other more complex types, but it would be helpful enough for my use case if it's able to do what is showed above.

**Comment by yuhr** [comment]
`Record` runtype has `fields` property which contains the original information of passed schema, and every runtype has `tag` property, for example: `Record(...).tag === 'record'`. Doesn't it fit your needs?

**Comment by SIGUSR97** [comment]
> `Record` runtype has `fields` property which contains the original information of passed schema, and every runtype has `tag` property, for example: `Record(...).tag === 'record'`. Doesn't it fit your needs?

yeah... I guess I will use that for now. But I think this could still be useful for other complex types like optional types and union types. And there really should be a proper documentation that goes beyond what was showed in the examples since they only cover the most basic use cases, and any more than that I would have to dig in to the source code and definitions...

**Comment by yuhr** [comment]
`Optional` has `undelying`, `Union` has `alternatives`, `Intersect` has `intersectees` etc., though these must be documented as well. Also maybe #217 could help that.

BTW I think it's reasonable idea to make helper methods (e.g. `optional`, `asPartial`, ...) **non-enumerable** properties. This way we could use simply `JSON.stringify(runtype)` in order to genuinely "serialize" runtypes into JSON.

**Comment by yuhr** [comment]
Built-in functions of runtypes will be all non-enumerable in v7, so this is resolved.
